src/components/common/CountrySelect.tsx

```diff
@@ -0,0 +1,51 @@
+import { LocationOnOutlined as LocationOnOutlinedIcon } from '@mui/icons-material';
+import { Autocomplete, Box, FormHelperText, TextField } from '@mui/material';
+import { useEffect } from 'react';
+import { ControllerRenderProps } from 'react-hook-form';
+
+import { COUNTRIES } from '@/constants';
+import { CreatePostType } from '@/types/post';
+
+interface LocationProps {
+  isSelected: boolean;
+  name: ControllerRenderProps<CreatePostType, 'country.name'>;
+}
+
+const CountrySelect = ({ isSelected, name }: LocationProps) => {
+  useEffect(() => {
+    if (name) {
+      isSelected ? name.onChange('대한민국') : name.onChange('');
+    }
+  }, [isSelected]);
```

Comment:
ask;

```suggestion
  }, [isSelected, name]);
```

`useEffect` 의존성에 `name` props를 추가하지 않은 이유가 있나요?

Reply:
저번에도 말씀드린것처럼 name을 주게되면, 해외를 선택후
나라를 입력할때 해당 함수가 호출되기때문에 name.onChange('')떄문에 입력이 불가능합니다